source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -1046,6 +1047,31 @@ message TransformationPushIdThroughVariable {
 
 }
 
+message TransformationRemoveParameters {
```

Comment:
I think TransformationReplaceParameterWithGlobal would be a better name.  Or TransformationPromoteParameterToGlobal.

Reply:
@Vasniktel and @paulthomson I would love to discuss the following comment:

Related to this, I think it would be nicer to do this one parameter at a time, rather than to remove several parameters simultaneously.

I suggest you have these fields:

- `uint32 parameter_id` -- the id of the formal parameter to be promoted.  Because ids are unique in SPIR-V this suffices to identify the function and its parameter; there is no need to have the function id.  Furthermore, this eliminates the problem of needing to worry about counting which parameter is being manipulated.

- `uint32 fresh_id`: for non-pointer parameters this is non-zero, and the new global variable has this id; the load from that variable gets the original parameter id.  For pointer parameters `fresh_id` is zero because the original parameter id can be used as the id for the global variable.

- `uint32 initializer_id`: the id with which to initialize the global variable)

I suggest we do *not* have the new function type as part of this transformation, and that the transformation adds the necessary function type on demand.

Reply:
Partially fixed in cc488d5.
> I suggest we do not have the new function type as part of this transformation, and that the transformation adds the 
> necessary function type on demand.

needs clarification.

Reply:
Fixed in 7b31019.